Assets/Scripts/SceneManageMent/ScoreTracker.cs

```diff
@@ -133,18 +133,34 @@ public void AddToScore(int points)
     /// <summary>Updates this class's Energy to the bike's energy.</summary>
     private void UpdateUIEnergy()
     {
-        Energy = bike.Energy;
+        if (bike == null)
+        {
+            Debug.Log("There is no bike.");
+        }
+        else
+        {
+            Energy = bike.Energy;
+        }
+
     }
 
     /// <summary>Loads the gameover screen.</summary>
     /// <param name="survivedEvent">True if the player beat the level.</param>
     private void EndGame(bool survivedEvent)
     {
         PlayerDataObject.survivedEvent = survivedEvent;
+        if (Object.FindObjectOfType<DLevel>() == null)
```

Comment:
Could you please call this method only once and store the result in a variable? This is a pretty costly method to call
```suggestion
DLevel dLevel = Object.FindObjectOfType<DLevel>();
        if (dLevel == null)
```